openvdb_points/tools/PointDataGrid.h

```diff
@@ -895,6 +905,25 @@ PointDataLeafNode<T, Log2Dim>::memUsage() const
     return BaseLeaf::memUsage() + mAttributeSet->memUsage();
 }
 
+
+////////////////////////////////////////
+
+
+template <typename PointDataTreeT>
+inline void
+makeDescriptorUnique(PointDataTreeT& tree)
+{
+    typename PointDataTreeT::LeafIter leafIter = tree.beginLeaf();
+    if (!leafIter)  return;
+
+    const AttributeSet::Descriptor& descriptor = leafIter->attributeSet().descriptor();
+    AttributeSet::Descriptor::Ptr newDescriptor(new AttributeSet::Descriptor(descriptor));
+    for (; leafIter; ++leafIter) {
+        leafIter->renameAttributes(descriptor, newDescriptor);
```

Comment:
I did think about this, the use case is primarily renaming (I think), but it does also just handle swapping of the descriptor so perhaps swapDescriptor() is a better name and renaming is more a special case of this. In fact, yeah think that makes more sense, I'll change it.